tests/validation/test_free_tier_compliance.py

```
@@ -0,0 +1,433 @@
+        """Test that rate limit exceptions contain correct information."""
+        limiter, mock_redis = rate_limiter
+        
+        # Set up rate limit exceeded condition
+        mock_redis.set_count("flash", "rpm", 10)
+        
+        try:
+            await limiter.execute_with_protection(
+                "gemini-2.5-flash",
+                lambda: "success"
+            )
+            assert False, "Should have raised RateLimitExceededException"
```

Ruff (0.11.9)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)